docs/development/Whats-Going-On-With-Movement.md

```diff
@@ -0,0 +1,69 @@
+<h1>Movement</h1>
+
+
+There are 2 types of movement
+
+1. Tile movement
+2. Newtonian movement
+
+
+<h1>Tile movement</h1>
+
+TryTilePush Is the initial Step into tile pushing, This does all the Checks with can move, 
+And then goes into ForceTilePush, This allows you to do some cheeky stuff like pushing into walls, 
+But also allows you to skip validation checks.
+
+Once this is done it sets SetLocalTarget, and adds AnimationUpdateMe to the Update manager,
+this function handles moving towards LocalTarget, This will move the game object towards the target until reached,
+it has options to apply momentum once it has reached the location
+
+
+
+<h2>Newtonian movement</h2>
+
+newtonianMovement Is the vector2 xy of the current momentum of the object, 
+All the calculations for it are handled in side of FlyingUpdateMe, 
+There are two extra attributes Important to momentum,
+
+
+airTime = How long the object will travel with no drag
+
+slideTime = How long the object will travel with reduced drag
+
+FlyingUpdateMe Also handles the collisions and bouncing for objects.
+
+<h2>Notes</h2>
+
+IsStickyMovement changes how The object,
+ Calculates if it has gravity, ( it checks all adjacent tiles Instead of just underneath it),
+ And also brings the momentum to a stop when it reaches the threshold of maximumStickSpeed,
+
+<h2>MovementSynchronisation</h2>
```

Comment:
This section is incomprehensible to me, I think it may be better to just leave a TODO in there.